framework/src/engine/consensus/consensus.ts

```diff
@@ -549,11 +549,13 @@ export class Consensus {
 			});
 			await this._executeValidated(block);
 
+			// Since legacy property is optional we don't need to send it here
 			this._network.applyNodeInfo({
 				height: block.header.height,
 				lastBlockID: block.header.id,
 				maxHeightPrevoted: block.header.maxHeightPrevoted,
 				blockVersion: block.header.version,
+				legacy: [], // TODO: call legacyChainHandler to get the updated value https://github.com/LiskHQ/lisk-sdk/issues/7503
```

Comment:
It would be nice if consensus don't need to know about `legacyChainHandler`

Reply:
yeah I'm just thinking how to manage this, maybe within the network we store `nodeInfo` and every time we call `applyNodeInfo(<Partial<NodeInfo>>)` we update the passed properties and send the whole nodeInfo to the network

Reply:
yea maybe that would be better approach than adding dependency